pkg/kv/kvserver/replica_consistency.go

```diff
@@ -406,21 +386,11 @@ func (r *Replica) runConsistencyCheck(
 	return nil, errors.New("could not collect checksum from any replica")
 }
 
-func (r *Replica) gcOldChecksumEntriesLocked(now time.Time) {
-	for id, val := range r.mu.checksums {
-		// The timestamp is valid only if set.
-		if !val.gcTimestamp.IsZero() && now.After(val.gcTimestamp) {
-			delete(r.mu.checksums, id)
-		}
-	}
-}
-
-// getReplicaChecksum returns replicaChecksum tracker for the given ID, and
-// whether it is still active (i.e. has a zero GC timestamp).
-func (r *Replica) getReplicaChecksum(id uuid.UUID, now time.Time) (*replicaChecksum, bool) {
+// getReplicaChecksum returns replicaChecksum tracker for the given ID. The
+// caller must call delReplicaChecksum when finished working on this tracker.
+func (r *Replica) getReplicaChecksum(id uuid.UUID) *replicaChecksum {
```

Comment:
How about returning a cleanup closure instead, to make this requirement more explicit? E.g.:

```go
c, cleanup := r.getReplicaChecksum()
defer cleanup()
```

We could also consider renaming to something like `trackReplicaChecksum` to make it clearer that this isn't a simple read operation. No strong opinion either way, your call.

Reply:
Nice idea, makes the contract self-explanatory and removes the possibility of passing in a wrong pair of `(*replicaChecksum, id)`. Done.